### DISCUSSION TITLE
Do I need jquery for cytoscape?

### DISCUSSION THREAD

**Discussion by kultigin** [cytoscape/cytoscape.js#2904] [Q&A]
I am using cytoscape for mapping relational data. In the same project I am using bootstrap but since the framework is angular I can get rid of jquery for bootstrap. but does cytoscape use it? Or Is it for popper and tippy?

**Comment by igordata** [comment]
Cytoscape doesn't need jQuery at all. It just has a method `.$()` allowing to use similar to jQuery syntax to select elements.

**Reply by kultigin** [reply]
Ok thanks.Thats good.  It looks like the coda layout behaves erratically without jquery though. Since I am using ng-bootstrap I wanted to get rid of jquery. There is popper and tippy too but popper doesn't need jquery either as far as I know.

**Reply by igordata** [reply]
I use elkjs for layouts. It's pretty good and doesn't require jQuery. Works fine serverside and on a client.
